Can't create a Custom ClickEvent without a BinaryTagHolder

### Expected behavior

Create a ClickEvent without a payload, without crashing, as specified by the API

### Observed/Actual behavior

I saw tons of errors in my logs :(

### Steps/models to reproduce

```java
Component component = Component.text("Helo!")
    .clickEvent(ClickEvent.custom(Key.key("milo", "bilo")));
player.sendMessage(component);
```

Logs:
```
[22:26:10 ERROR]: Error sending packet clientbound/minecraft:system_chat
io.netty.handler.codec.EncoderException: Failed to encode packet 'clientbound/minecraft:system_chat'
        at net.minecraft.network.codec.IdDispatchCodec.encode(IdDispatchCodec.java:63) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at net.minecraft.network.codec.IdDispatchCodec.encode(IdDispatchCodec.java:13) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at net.minecraft.network.PacketEncoder.encode(PacketEncoder.java:27) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at net.minecraft.network.PacketEncoder.encode(PacketEncoder.java:12) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at io.netty.handler.codec.MessageToByteEncoder.write(MessageToByteEncoder.java:107) ~[netty-codec-base-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:800) ~[netty-transport-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:737) ~[netty-transport-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:113) ~[netty-codec-base-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:800) ~[netty-transport-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:737) ~[netty-transport-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.channel.ChannelOutboundHandlerAdapter.write(ChannelOutboundHandlerAdapter.java:113) ~[netty-transport-4.2.15.Final.jar:4.2.15.Final]
        at net.minecraft.network.Connection$2.write(Connection.java:685) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:800) ~[netty-transport-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:737) ~[netty-transport-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1036) ~[netty-transport-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.channel.Channel.write(Channel.java:267) ~[netty-transport-4.2.15.Final.jar:4.2.15.Final]
        at net.minecraft.network.Connection.doSendPacket(Connection.java:469) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at net.minecraft.network.Connection.lambda$sendPacket$0(Connection.java:449) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:148) ~[netty-common-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:141) ~[netty-common-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:535) ~[netty-common-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.channel.SingleThreadIoEventLoop.run(SingleThreadIoEventLoop.java:201) ~[netty-transport-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:1195) ~[netty-common-4.2.15.Final.jar:4.2.15.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.2.15.Final.jar:4.2.15.Final]
        at java.base/java.lang.Thread.run(Thread.java:1516) ~[?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "net.kyori.adventure.nbt.api.BinaryTagHolder.get(net.kyori.adventure.util.Codec)" because "api" is null
        at io.papermc.paper.adventure.AdventureCodecs.lambda$static$1(AdventureCodecs.java:88) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at com.mojang.serialization.Encoder$2.encode(Encoder.java:39) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.Codec$2.encode(Codec.java:79) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.Encoder.encodeStart(Encoder.java:14) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.FieldEncoder.encode(FieldEncoder.java:25) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.MapCodec$2.encode(MapCodec.java:83) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.RecordCodecBuilder$Instance$4.encode(RecordCodecBuilder.java:224) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.RecordCodecBuilder$2.encode(RecordCodecBuilder.java:112) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.KeyDispatchCodec.encode(KeyDispatchCodec.java:71) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.MapCodec$MapCodecCodec.encode(MapCodec.java:164) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.Encoder.encodeStart(Encoder.java:14) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.OptionalFieldCodec.encode(OptionalFieldCodec.java:44) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.OptionalFieldCodec.encode(OptionalFieldCodec.java:17) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.RecordCodecBuilder$Instance$6.encode(RecordCodecBuilder.java:298) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.RecordCodecBuilder$Instance$6.encode(RecordCodecBuilder.java:295) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.RecordCodecBuilder$2.encode(RecordCodecBuilder.java:112) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.RecordCodecBuilder$Instance$6.encode(RecordCodecBuilder.java:298) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.RecordCodecBuilder$2.encode(RecordCodecBuilder.java:112) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.MapCodec$MapCodecCodec.encode(MapCodec.java:164) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.EitherCodec.lambda$encode$4(EitherCodec.java:35) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.datafixers.util.Either$Right.map(Either.java:99) ~[paper-26.2.jar:?]
        at com.mojang.serialization.codecs.EitherCodec.encode(EitherCodec.java:33) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.codecs.EitherCodec.encode(EitherCodec.java:11) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.Encoder$1.encode(Encoder.java:25) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.Codec$2.encode(Codec.java:79) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.Codec$RecursiveCodec.encode(Codec.java:224) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.Encoder.encodeStart(Encoder.java:14) ~[datafixerupper-10.0.21.jar:?]
        at io.papermc.paper.adventure.WrapperAwareSerializer.serialize(WrapperAwareSerializer.java:37) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at io.papermc.paper.adventure.WrapperAwareSerializer.serialize(WrapperAwareSerializer.java:13) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at io.papermc.paper.adventure.AdventureComponent.deepConverted(AdventureComponent.java:26) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at io.papermc.paper.adventure.AdventureComponent.getSiblings(AdventureComponent.java:57) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at net.minecraft.network.chat.Component.tryCollapseToString(Component.java:69) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at net.minecraft.network.chat.ComponentSerialization.lambda$createCodec$4(ComponentSerialization.java:182) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at com.mojang.serialization.Encoder$1.encode(Encoder.java:25) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.Codec$2.encode(Codec.java:79) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.Codec$RecursiveCodec.encode(Codec.java:224) ~[datafixerupper-10.0.21.jar:?]
        at com.mojang.serialization.Encoder.encodeStart(Encoder.java:14) ~[datafixerupper-10.0.21.jar:?]
        at net.minecraft.network.chat.ComponentSerialization$1.encode(ComponentSerialization.java:57) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at net.minecraft.network.chat.ComponentSerialization$1.encode(ComponentSerialization.java:44) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at net.minecraft.network.codec.StreamCodec$8.encode(StreamCodec.java:152) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at net.minecraft.network.codec.StreamCodec$5.encode(StreamCodec.java:94) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at net.minecraft.network.codec.StreamCodec$5.encode(StreamCodec.java:84) ~[paper-26.2.jar:26.2-60-1cb58fb]
        at net.minecraft.network.codec.IdDispatchCodec.encode(IdDispatchCodec.java:58) ~[paper-26.2.jar:26.2-60-1cb58fb]
        ... 24 more
```

### Plugin and Datapack List

The plugin doing this and PolarPaper

### Paper version

Paper version 26.2-60-main@1cb58fb (2026-07-12T16:08:02Z) (Implementing API version 26.2.build.60-beta)

### Other

It does work fine if I just provide a payload, but that was not what I wanted

--- a/paper-server/src/main/java/io/papermc/paper/adventure/AdventureCodecs.java
+++ b/paper-server/src/main/java/io/papermc/paper/adventure/AdventureCodecs.java
@@ -149,8 +149,10 @@ public final class AdventureCodecs {
     ).apply(instance, ClickEvent::showDialog)));
     static final MapCodec<ClickEvent> CUSTOM_CODEC = mapCodec((instance) -> instance.group(
         KEY_CODEC.fieldOf("id").forGetter(a -> ((ClickEvent.Payload.Custom) a.payload()).key()),
-        BINARY_TAG_HOLDER_CODEC.fieldOf("payload").forGetter(a -> ((ClickEvent.Payload.Custom) a.payload()).nbt())
-    ).apply(instance, ClickEvent::custom));
+        BINARY_TAG_HOLDER_CODEC.optionalFieldOf("payload").forGetter(a -> Optional.ofNullable(((ClickEvent.Payload.Custom) a.payload()).nbt()))
+    ).apply(instance, (key, binaryTagHolder) -> {
+        return ClickEvent.custom(key, binaryTagHolder.orElse(null));
+    }));
 
     static final ClickEventType OPEN_URL_CLICK_EVENT_TYPE = new ClickEventType(OPEN_URL_CODEC, "open_url");
     static final ClickEventType OPEN_FILE_CLICK_EVENT_TYPE = new ClickEventType(OPEN_FILE_CODEC, "open_file");

